README.md

```diff
@@ -4,46 +4,81 @@
+
+Function `generate_png_scaled_custom` performs the scaling with custom parameters, and requires:  
+- `&[u8]` slice  
+- **target** identicon size in pixels (`u8` - it is for small identicons, after all)  
+- scaling factor (`u8`), how much larger the larger png actually is  
+- filter ([`FilterType`](https://docs.rs/image/latest/image/imageops/enum.FilterType.html)) used for image resize  
+
+The scaling factor reasonable values are in range `[4..=8]`, below it the pixelation persists, above it the images are not visibly improving anymore, and may even seem blurry.  
+
+All filters produce reasonable results, except `FilterType::Nearest` that yields visibly distorted images and therefore is not recommended.  
+
+Function `generate_png_scaled_default` performs the scaling with default scaling parameters (scaling factor `5` and filter `FilterType::Lanczos3`) for image with default Signer identicon size (30 pix), and requires only:  
+- `&[u8]` slice  
+If somehow the generation of the identicon fails, function outputs default-sized (30x30) transparent png image, i.e. it never produces an error.  
+
+Function `generate_png_with_colors` is similar to `generate_png`, but accepts identicon colors directly, and does not generate color set itself. This is intended mainly for tests. Function `generate_png_with_colors` requires:  
+- `[[u8; 4]; 19]` 19-element set of colors in RGBA format  
+- target image size in pixels (`u16`)  
+
+Function `generate_png_scaled_custom_with_colors` is similar to `generate_png_scaled_custom`, but accepts identicon colors directly, and does not generate color set itself. This is intended mainly for tests. Function `generate_png_scaled_custom_with_colors` requires:  
+- `[[u8; 4]; 19]` 19-element set of colors in RGBA format  
+- target identicon size in pixels (`u8`)  
+- scaling factor (`u8`)  
+- filter ([`FilterType`](https://docs.rs/image/latest/image/imageops/enum.FilterType.html)) used for image resize  
+
+
+## SVG
+
+Feature `"vec"` (enabled by default) enables infallible generation of identicon pictures in svg format. Since svg is a vector format, no image size parameters are needed.
+
+Function `generate_svg` reqires only `&[u8]` input slice.  
+
+Function `generate_svg_with_colors` uses pre-set colors and is intended mainly for tests. It requires only the color set (`[[u8; 4]; 19]` 19-element set of colors in RGBA format).  
 
 
 ## Tests and Examples
 
-Currently there are tests to check if the color sets calculated for Alice and Bob are identical to the colors in the corresponding well-known icons.  
+Tests in `colors.rs` module check if the color sets calculated for Alice and Bob are identical to the colors in the corresponding well-known icons.  
 
-Examples are plotting both .png and .svg icons for base58 westend Alice address, and hex westend genesis hash.
+Doc tests in `lib.rs` produce various test pics, both png (through different functions and parameters) and svg.  
```

Comment:
Use backticks for `png` and `svg`. Adding `cargo-spellcheck` as a CI stage might make sense as well.